monai/apps/deepgrow/transforms.py

```diff
@@ -477,3 +489,436 @@ def __call__(self, data):
+            metadata `image_meta_dict` dictionary's `affine` field.
+    """
+
+    def __init__(
+        self,
+        ref_image,
+        guidance: str = "guidance",
+        foreground: str = "foreground",
+        background: str = "background",
+        axis: int = 0,
+        depth_first: bool = True,
+        dimensions: int = 2,
+        slice: str = "slice",
+        meta_key_postfix: str = "meta_dict",
+    ):
+        self.ref_image = ref_image
+        self.guidance = guidance
+        self.foreground = foreground
+        self.background = background
+        self.axis = axis
+        self.depth_first = depth_first
+        self.dimensions = dimensions
+        self.slice = slice
+        self.meta_key_postfix = meta_key_postfix
+
+    def _apply(self, pos_clicks, neg_clicks, factor, slice_num):
+        pos = neg = []
+
+        if self.dimensions == 2:
+            points = list(pos_clicks)
+            points.extend(neg_clicks)
+            points = np.array(points)
+
+            slices = np.unique(points[:, self.axis]).tolist()
+            slice_idx = slices[0] if slice_num is None else next(x for x in slices if x == slice_num)
+
+            if len(pos_clicks):
+                pos_clicks = np.array(pos_clicks)
+                pos = (pos_clicks[np.where(pos_clicks[:, self.axis] == slice_idx)] * factor)[:, 1:].astype(int).tolist()
+            if len(neg_clicks):
+                neg_clicks = np.array(neg_clicks)
+                neg = (neg_clicks[np.where(neg_clicks[:, self.axis] == slice_idx)] * factor)[:, 1:].astype(int).tolist()
+
+            guidance = [pos, neg, slice_idx]
+        else:
+            if len(pos_clicks):
+                pos = np.multiply(pos_clicks, factor).astype(int).tolist()
+            if len(neg_clicks):
+                neg = np.multiply(neg_clicks, factor).astype(int).tolist()
+            guidance = [pos, neg]
+        return guidance
+
+    def __call__(self, data):
+        d = dict(data)
+        meta_dict_key = f"{self.ref_image}_{self.meta_key_postfix}"
+        if meta_dict_key not in d:
+            raise RuntimeError(f"Missing meta_dict {meta_dict_key} in data!")
+        if "spatial_shape" not in d[meta_dict_key]:
+            raise RuntimeError('Missing "spatial_shape" in meta_dict!')
+        original_shape = d[meta_dict_key]["spatial_shape"]
+        current_shape = list(d[self.ref_image].shape)
+
+        if self.depth_first:
+            if self.axis != 0:
+                raise RuntimeError("Depth first means the depth axis should be 0.")
+            # in here we assume the depth dimension was in the last dimension of "original_shape"
+            original_shape = np.roll(original_shape, 1)
+
+        factor = np.array(current_shape) / original_shape
+
+        fg_bg_clicks = []
+        for key in [self.foreground, self.background]:
+            clicks = json.loads(d[key]) if isinstance(d[key], str) else d[key]
```

Comment:
the `json.loads` should be done beforehand. could you remove any of the configuration related logic in case I missed them?

Reply:
@YuanTingHsieh we can remove this check.. and assumes users to provide an array in the dict for background/foreground clicks/input.

Reply:
done